drivers/net/macsec.c

```diff
@@ -2967,14 +2967,22 @@ static netdev_tx_t macsec_start_xmit(struct sk_buff *skb,
 	return ret;
 }
 
-#define MACSEC_FEATURES \
+#define SW_MACSEC_FEATURES \
 	(NETIF_F_SG | NETIF_F_HIGHDMA | NETIF_F_FRAGLIST)
 static struct lock_class_key macsec_netdev_addr_lock_key;
 
+/* If h/w offloading is enabled, use real device features save for
+ *   VLAN_FEATURES - they require additional ops
+ *   HW_MACSEC - no reason to report it
+ */
+#define REAL_DEV_FEATURES(dev) \
+	((dev)->features & ~(NETIF_F_VLAN_FEATURES | NETIF_F_HW_MACSEC))
```

Comment:
In practice I think we may redirect add_vid/del_vid callbacks to the real_dev.
I however found no similar practice in any of other wrapping netdev drivers, so not sure if that could be accepted in upstream. We may try doing RFC for such a thing later.

Reply:
Sounds good to me.